bcipy/gui/BCInterface.py

```diff
@@ -327,6 +328,54 @@ def check_input(self) -> bool:
             return False
         return True
 
+    def user_id_validator(self) -> bool:
+        """Check User ID Requirements. User ID must follow the following Requirements:
+
+        1. Maximum length of 25 letters or numbers
+        2. Minimum length of at least one letter or  number
+        3. No special characters
+        4. No spaces
+        """
+
+        # Update based on current inputs
+        self.user = self.user_input.currentText()
+        allowed_characters = re.compile('[^0-9a-zA-Z]+')
+
+        # Check the length of minimum one character up to 25 characters
+        try:
+            if len(self.user) == 0 or len(self.user) > 25:
```

Comment:
Consider assigning a `length_user` variable since you call it twice, and you could then append it in the message of the alert using f strings,

`f'User ID must be of at least one or up to 25. Your current ID is {length_user} characters.'`